### PR TITLE
TP: adds the ability to delete a job (aka invalidation request)

### PR DESCRIPTION
**Pull request by mitchell852**
- [x] This PR is related to #2030

Adds the ability in TP to delete an invalidate content request job. User is restricted to only deleting jobs within their tenancy, otherwise, only jobs associated with delivery services within their tenancy.

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?

- Navigate to https://traffic-portal.tld/#!/jobs. Delete a job by right-clicking on a job row. Note the delivery service of the job. Note the cdn of the delivery service.
- Navigate to https://traffic-portal.tld/#!/cdns/{cdn-ID}/servers. Note that the "Reval Pending" flag for all servers (except for OFFLINE or PRE_PROD) have been set to true showing a clock icon.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [ ] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by ocket8888** [comment]
I'm pretty sure I wanted to do this when translating the endpoint from Perl to Go and the ML decided we shouldn't delete jobs.

**Comment by ocket8888** [comment]
Nevermind that comment; it's not true. The DELETE handler already exists in the API, which is sufficient proof that it wasn't shot down on the mailing list.

File: traffic_portal/app/src/common/modules/table/jobs/TableJobsController.js [hunk 1]
```diff
@@ -17,10 +17,134 @@
...
+
+	/** All of the jobs - startTime fields converted to actual Dates and derived expires field from TTL */
+	$scope.jobs = jobs.map(
+		function(x) {
+			// need to convert this to a date object for ag-grid filter to work properly
+			x.startTime = new Date(x.startTime.replace("+00", "Z"));
+
+			// going to derive the expires date from start + TTL (hours). Format: TTL:24h
+			let ttl = x.parameters.slice('TTL:'.length, x.parameters.length-1);
+			x.expires = new Date(x.startTime.getTime() + ttl*3600*1000);
```
[the remark this answers is not in the document]
Comment (mitchell852) [reply traffic_portal/app/src/common/modules/table/jobs/TableJobsController.js:80]: added. good catch.

File: traffic_portal/app/src/common/modules/table/jobs/TableJobsController.js [hunk 2]
```diff
@@ -17,10 +17,134 @@
...
+		enableCellTextSelection: true,
+		defaultColDef: {
+			filter: true,
+			sortable: true,
+			resizable: true,
+			tooltip: defaultTooltip
+		},
+		rowClassRules: {
+			'active-job': function(params) {
+				return new Date(params.data.expires) > new Date();
```
Comment (ocket8888) [inline traffic_portal/app/src/common/modules/table/jobs/TableJobsController.js:101]: expires is already a Date according to the `.map` you did before, so this construction should be redundant.

File: traffic_portal/app/src/common/modules/table/jobs/table.jobs.tpl.html [hunk 3]
```diff
@@ -22,34 +22,53 @@
...
-            <button class="btn btn-primary" title="Create Invalidation Request" ng-click="createJob()"><i class="fa fa-plus"></i></button>
-            <button class="btn btn-default" title="Refresh" ng-click="refresh()"><i class="fa fa-refresh"></i></button>
+        <div class="pull-right">
+            <div class="form-inline" role="search">
+                <input id="quickSearch" name="quickSearch" type="search" class="form-control text-input" placeholder="Quick search..." ng-model="quickSearch" ng-change="onQuickSearchChanged()" aria-label="Search"/>
+                <div class="input-group text-input">
+                    <span class="input-group-addon">
+                        <label for="pageSize">Page size</label>
+                    </span>
+                    <input id="pageSize" name="pageSize" type="number" class="form-control" placeholder="100" ng-model="pageSize" ng-change="onPageSizeChanged()" aria-label="Page Size"/>
```
Comment (ocket8888) [inline traffic_portal/app/src/common/modules/table/jobs/table.jobs.tpl.html:32]: nit but you don't need an aria-label if you have an actual label. I don't think it's hurting anything, though.

**Review by ocket8888 — COMMENTED** [review]
Confirmed that I can now delete jobs

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**